test_responses.py

```diff
@@ -334,6 +335,20 @@ def run():
     assert_reset()
 
 
+def test_match_querystring_missing_key():
+    @responses.activate
+    def run():
+        responses.add(responses.GET, "http://example.com?foo=1&bar=2", body=b"test")
+        with pytest.raises(ConnectionError):
+            requests.get("http://example.com/?foo=1&baz=2")
+
+        with pytest.raises(ConnectionError):
+            requests.get("http://example.com/?bar=2&fez=1")
+
+    run()
+    assert_reset()
```

Comment:
I think it would be good to have a test for query string params count like: `http://example.com/?foo=1`.
Also, you can use [pytest parametrize](https://docs.pytest.org/en/latest/parametrize.html).
```suggestion
@pytest.mark.parametrize(
    "test_url",
    [
        "http://example.com/?foo=1&baz=2",
        "http://example.com/?bar=2&fez=1",
        "http://example.com/?foo=1",
    ]
)
def test_match_querystring_missing_key(test_url):
    @responses.activate
    def run():
        responses.add(responses.GET, "http://example.com?foo=1&bar=2", body=b"test")
        with pytest.raises(ConnectionError):
            requests.get(test_url)

    run()
    assert_reset()

```

Reply:
This is already tested elsewhere. The goal of this commit was to handle the case where the keys don't match and everything else does; ultimately changing the code to a straight comparison between the two lists removes the need for this edge case anyway.